File: ruby/001.rb
#Multiples of 3 and 5
#Problem 1 
#If we list all the natural numbers below 10 that are multiples of 3 or 5, we get 3, 5, 6 and 9. The sum of these multiples is 23.
#
#Find the sum of all the multiples of 3 or 5 below 1000.

numbers = (1..999)

sum = numbers.inject(0) do |acc, num| 
  (num % 3 == 0 or num % 5 == 0) ? acc + num : acc 
end

puts "Sum = ", sum

File: elixir/001.ex
# If we list all the natural numbers below 10 that are multiples of 3 or 5, we get 3, 5, 6 and 9. The sum of these multiples is 23.
# Find the sum of all the multiples of 3 or 5 below 1000.

import Kernel
import Enum

defmodule Euler do
  def reduceFunction(number, accumulator) do
    if (rem(number, 3) == 0 || rem(number, 5) == 0), do: number + accumulator, else: accumulator
  end
end

IO.puts reduce(to_list(1..999), 0, &Euler.reduceFunction/2)
